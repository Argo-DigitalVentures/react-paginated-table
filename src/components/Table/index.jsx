import React, { PropTypes } from 'react';

import TheadComponent from '../Thead';
import TbodyComponent from '../Tbody';
import TrComponent from '../Tr';
import ThComponent from '../Th';
import TdComponent from '../Td';

import {
  styleType,
  classNameType,
  dataType,
  columnType,
  sortType,
} from '../../propTypes';

function Table(props) {
  const {
    data,
    columns,
    sorts,
    handleHeaderClick,
    TheadComponent,
    TbodyComponent,
    TrComponent,
    ThComponent,
    TdComponent,
    theadStyle,
    tbodyStyle,
    trStyle,
    thStyle,
    tdStyle,
    theadClassName,
    tbodyClassName,
    trClassName,
    thClassName,
    tdClassName,
    ...rest
  } = props;
  return (
    <table {...rest}>
      <TheadComponent
        style={theadStyle}
        className={theadClassName}
        data={data}
        columns={columns}
        sorts={sorts}
        handleHeaderClick={handleHeaderClick}
        ThComponent={ThComponent}
        thStyle={thStyle}
        thClassName={thClassName}
      />
      <TbodyComponent
        style={tbodyStyle}
        className={tbodyClassName}
        data={data}
        columns={columns}
        TrComponent={TrComponent}
        TdComponent={TdComponent}
        trStyle={trStyle}
        tdStyle={tdStyle}
        trClassName={trClassName}
        tdClassName={tdClassName}
      />
    </table>
  );
}

Table.defaultProps = {
  style: {},
  className: '',
  TheadComponent,
  TbodyComponent,
  TrComponent,
  ThComponent,
  TdComponent,
  theadStyle: {},
  tbodyStyle: {},
  trStyle: {},
  thStyle: {},
  tdStyle: {},
  theadClassName: '',
  tbodyClassName: '',
  trClassName: '',
  thClassName: '',
  tdClassName: '',
};

Table.propTypes = {
  data: dataType.isRequired,
  columns: PropTypes.arrayOf(columnType).isRequired,
  sorts: PropTypes.arrayOf(sortType).isRequired,
  handleHeaderClick: PropTypes.func.isRequired,
  style: styleType,
  className: classNameType,
  TheadComponent: PropTypes.func,
  TbodyComponent: PropTypes.func,
  TrComponent: PropTypes.func,
  ThComponent: PropTypes.func,
  TdComponent: PropTypes.func,
  theadStyle: styleType,
  tbodyStyle: styleType,
  trStyle: styleType,
  thStyle: styleType,
  tdStyle: styleType,
  theadClassName: classNameType,
  tbodyClassName: classNameType,
  trClassName: classNameType,
  thClassName: classNameType,
  tdClassName: classNameType,
};

export default Table;
