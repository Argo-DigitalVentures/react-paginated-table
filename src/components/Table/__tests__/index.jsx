import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import Table from '../';

function createFakeComponent() {
  return () => null;
}

describe('<Table />', () => {
  it('should accept className, style, and other DOM props', () => {
    const fakeFn = () => null;
    const TheadComponent = createFakeComponent();
    const TbodyComponent = createFakeComponent();
    const TrComponent = createFakeComponent();
    const ThComponent = createFakeComponent();
    const TdComponent = createFakeComponent();
    const tree = shallow(
      <Table
        data={[]}
        columns={[]}
        sorts={[]}
        handleHeaderClick={fakeFn}
        style={{
          color: 'green',
        }}
        className="test"
        TheadComponent={TheadComponent}
        TbodyComponent={TbodyComponent}
        TrComponent={TrComponent}
        ThComponent={ThComponent}
        TdComponent={TdComponent}
        theadStyle={{
          color: 'red',
        }}
        tbodyStyle={{
          color: 'blue',
        }}
        trStyle={{
          color: 'orange',
        }}
        thStyle={{
          color: 'purple',
        }}
        tdStyle={{
          color: 'gray',
        }}
        theadClassName="theadClassName"
        tbodyClassName="tbodyClassName"
        trClassName="trClassName"
        thClassName="thClassName"
        tdClassName="tdClassName"
      />,
    );
    expect(toJson(tree)).toMatchSnapshot();
  });
});
