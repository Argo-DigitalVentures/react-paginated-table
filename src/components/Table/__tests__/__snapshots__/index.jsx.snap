exports[`<Table /> should accept className, style, and other DOM props 1`] = `
<table
  className="test"
  style={
    Object {
      "color": "green",
    }
  }>
  <Component
    ThComponent={[Function]}
    className="theadClassName"
    columns={Array []}
    data={Array []}
    handleHeaderClick={[Function]}
    sorts={Array []}
    style={
      Object {
        "color": "red",
      }
    }
    thClassName="thClassName"
    thStyle={
      Object {
        "color": "purple",
      }
    } />
  <Component
    TdComponent={[Function]}
    TrComponent={[Function]}
    className="tbodyClassName"
    columns={Array []}
    data={Array []}
    style={
      Object {
        "color": "blue",
      }
    }
    tdClassName="tdClassName"
    tdStyle={
      Object {
        "color": "gray",
      }
    }
    trClassName="trClassName"
    trStyle={
      Object {
        "color": "orange",
      }
    } />
</table>
`;
